drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1010 @@
+/* fdc2x1x.c - Driver for the Texas Instruments FDC2X1X */
+
+/*
+ * Copyright (c) 2020 arithmetics ITS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_fdc2x1x
+
+#include <device.h>
+#include <sys/util.h>
+#include <logging/log.h>
+
+#include "fdc2x1x.h"
+
+LOG_MODULE_REGISTER(FDC2X1X, CONFIG_SENSOR_LOG_LEVEL);
+
+static int fdc2x1x_init_config(const struct device *dev);
+
+/**
+ * Read/Write from device.
+ * @param dev - The device structure.
+ * @param reg - The register address. Use FDC2X1X_REG_READ(x) or
+ *				FDC2X1X_REG_WRITE(x).
+ * @param data - The register data.
+ * @param length - Number of bytes being read
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_bus_access(const struct device *dev, uint8_t reg,
+			      void *data, size_t length)
+{
+	struct fdc2x1x_data *fdc2x1x_data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+
+	if (reg & FDC2X1X_READ) {
+		return i2c_burst_read(fdc2x1x_data->bus,
+				      cfg->i2c_addr,
+				      FDC2X1X_TO_I2C_REG(reg),
+				      (uint8_t *) data, length);
+	} else {
+		if (length <= 0) {
```

Comment:
What's going on here? The `length` argument is a `size_t`, which is unsigned, so this doesn't look right.

Should this be something like `if (length != 2)` instead? The `memcpy(buf + 1, data, sizeof(uint16_t));` line below makes it seem like exactly two bytes of `data` are being used in this case.

Reply:
your are right, my bad